### PR TITLE
Fix Jetpack redirection issue when the auth endpoint returns an invalid url

### LINKED ISSUE #38758: Fix Jetpack redirection error when Auth endpoint return an invalid URL
**Prerequisites (mark completed items with an [x]):**
- [ ] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [ ] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

**Issue Description:**

In the core profiler, the code does not account for a scenario where the authentication endpoint returns an invalid URL.

Please add a condition to verify the success flag of the endpoint and redirect users to the WooCommerce Home if the value is false.

### PR DESCRIPTION
**Pull request by moon0326**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #38758 

This PR handles a situation where the getJetpackAuth endpoint returns an invalid URL by redirecting users to WooCommerce Home. It also replaces `window.location.href` from `redirectToWooHome` with `navigateTo`.

### How to test the changes in this Pull Request:

- Make sure your env is not connected to Jetpack. Otherwise, it'll always redirect to Woo Home after the plugins page.

1. Use your local env that is not publicly accessible.
2. Install the `WooCommerce Beta Tester` plugin
3. Go to /wp-admin/tools.php?page=woocommerce-admin-test-helper
4. Enable the 'core-profiler' feature flag
5. Visit http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fsetup-wizard
6. Go to the plugins page.
7. Make sure to choose at least one plugin.
8. Open your browser's inspector
9. Click the Continue button.
10. After installation, you should be redirected to Woo Home since your local can't get a valid Jetpack auth URL.

Build a zip file with `pnpm run build:zip` and make a new JN site. Repeat the same steps on the JN site and confirm it redirects you to Jetpack Auth page.

### CODE REVIEW

**Review by chihsuan — APPROVED** [review]
Thanks for fixing this! LGTM and tested well. 🚢

File: plugins/woocommerce-admin/client/core-profiler/index.tsx [hunk 1]
```diff
@@ -322,10 +327,16 @@ const handleGeolocation = assign( {
...
+	navigateTo( {
+		url: getNewPath( {}, '/', {} ),
+	} );
+};
+
+const redirectToJetpackAuthPage = (
+	_context: CoreProfilerStateMachineContext,
+	event: { data: { url: string } }
+) => {
+	window.location.href = event.data.url;
```
Comment (chihsuan) [inline plugins/woocommerce-admin/client/core-profiler/index.tsx]:
```suggestion
	window.location.href = event.data.url + '&installed_ext_success=1';
```
Or we can add this param in backend

**Review by chihsuan — CHANGES_REQUESTED** [review]
Ah, Sorry, just noticed that `installed_ext_success` query param was accidentally removed.

**Review by adrianduffell — APPROVED** [review]
This tested well, thanks @moon0326 🚀

File: plugins/woocommerce-admin/client/core-profiler/index.tsx (hunk 1 above)
Comment (moon0326) [reply plugins/woocommerce-admin/client/core-profiler/index.tsx]: I think we can add it here.

**Review by chihsuan — APPROVED** [review]
🚀

**Pull request merged**